### ISSUE TITLE
Problem using Bincrafters Qt with *NIX systems

### ISSUE THREAD

**Issue by ShamanTcler** [conan-io/conan-docker-tools#190]
Seems a number of libs such as Qt depend up a version of "libxcb1" be installed. Doing a conan install, it fails with a reference to libxcb1.  It appears this package is not installed in the docker image, and the user of the image may not have privileges or knowledge to install it. 

This is similar to a previous issue: https://github.com/bincrafters/community/issues/452

Now could you say I should do it by creating a new docker image, sure.  But, I think this may come back to you as a reoccurring issue

Thanks for the hard work.

Regards
Carl

### Environment Details
* Conan Docker Tools image: docker.io/conanio/gcc49 /
                                               docker.io/conanio/clang9/

* Operating System: **Linux**
 

1. docker run -it docker.io/conanio/clang9 /bin/bash (or docker run -it docker.io/conanio/clang9 /bin/bash)
2. conan config set proxies.https=http://yupItWorks
3. conan config set proxies.http=http://yupItWorks
4. conan remote add bincrafters https://api.bintray.com/conan/bincrafters/public-conan false
5. conan install qt/5.9.8@bincrafters/stable --build -r bincrafters

The last command produces:

Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=clang
compiler.libcxx=libstdc++
compiler.version=9
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

qt/5.9.8@bincrafters/stable: Forced build from source
dpkg-query: no packages found matching libxcb1-dev
Running: sudo apt-get update
Err:1 http://security.ubuntu.com/ubuntu eoan-security InRelease
  Could not resolve 'security.ubuntu.com'
Err:2 http://archive.ubuntu.com/ubuntu eoan InRelease
  Could not resolve 'archive.ubuntu.com'
Err:3 http://archive.ubuntu.com/ubuntu eoan-updates InRelease
  Could not resolve 'archive.ubuntu.com'
Err:4 http://archive.ubuntu.com/ubuntu eoan-backports InRelease
  Could not resolve 'archive.ubuntu.com'
Reading package lists...
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/eoan/InRelease  Could not resolve 'archive.ubuntu.com'
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/eoan-updates/InRelease  Could not resolve 'archive.ubuntu.com'
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/eoan-backports/InRelease  Could not resolve 'archive.ubuntu.com'
W: Failed to fetch http://security.ubuntu.com/ubuntu/dists/eoan-security/InRelease  Could not resolve 'security.ubuntu.com'
W: Some index files failed to download. They have been ignored, or old ones used instead.
Running: sudo apt-get install -y --no-install-recommends libxcb1-dev
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libxcb1-dev
ERROR: qt/5.9.8@bincrafters/stable: Error in build_requirements() method, line 118
        installer.install(item + self._system_package_architecture())
        ConanException: Command 'sudo apt-get install -y --no-install-recommends libxcb1-dev' failed

**Comment by ericLemanissier** [comment]
qt recipe is trying to install libxcb1, but it fails because of a connection error to dpkg repository, are you sure that the proxies configuration is not messing it up ? I just tried your procedure without steps 2 and 3 and it worked.

**Comment by ShamanTcler** [comment]
Just tried it without proxies, same results. Yes I am behind corporate firewalls... and I HATE them, but I do like having the mortgage paid. 

I pretty much agree with you though this is a Proxy issue.

From what I read libxcb1 seems to be the X11 C interface. Is that common enough to include in the docker image?

**Comment by ericLemanissier** [comment]
it looks like it's the plan : https://github.com/conan-io/conan-center-index/issues/641#issuecomment-575184483 so we'll get there at some point. probably @uilianries has an idea on this ?

**Comment by uilianries** [comment]
Proxy requirements. But is under study yet.

About that error, is not an privilege error, but Ubuntu remote was dropped due EOL. Now it's okay, I just pointed to the backport. Anyway, the regular user is totally able installing any APT package.

**Comment by ShamanTcler** [comment]
Ok so let me see if I understand ... this docker container is based on ubuntu:trusty (https://wiki.ubuntu.com/Releases)  which went out of support in April of 2019 so trying to build on this image is likely to fail.... unless you have paid support.

I am running into this issue again as I am trying to add ssh to the container. So same "block" is affecting two changes.  (apt-get install openssh-server is giving me () failed: Bad record MAC)

Should I just be build my own container using say:  Ubuntu 18.04.4 LTS (Bionic Beaver) as opposed to trying to use the pre-built docker image from docker hub?

**Comment by uilianries** [comment]
We can't move it to another Ubuntu version because a simple matter: glibc version. It's totally incompatible, and will break ALL packages available in Conan Center. We old-release.ubuntu repo (vault) to get Trusty.

If you want to create a new image based on Bionic, it would be nice, but we can't accept because glibc version.

**Comment by ericLemanissier** [comment]
on my machine, `apt-get install openssh-server` is running fine inside conanio/gcc49 and conanio/clang9. Are you using the latest docker images (did you try `docker pull conanio/gcc49:latest` and `docker pull conanio/clang9:latest` ?

**Comment by ShamanTcler** [comment]
Uilianries, thanks for the insight on glibc, makes sense.

Eric I went back and made sure the latest tag was used, still not love. 

I decide to leave the corporate world behind for now, I went home and dusted off a laptop and installed Centos 8.  I was chugging along ready  to install dockers to replicate your experiance.  I then get hit with:

_"Red Hat seems to have somehow blocked the installation of containerd.io > 1.2.0-3.el7, which is a dependency of docker-ce. "_

This is just too much for a Friday afternoon... Since I live near Kentucky, I think it is time for a bourbon. If you guys were here, I would do my best to supply the adult beverage of your choice.

Thanks for your help
